cmd/dep/status.go

```diff
@@ -246,6 +378,21 @@ func runStatusAll(out outputter, p *dep.Project, sm *gps.SourceMgr) error {
 				PackageCount: len(proj.Packages()),
 			}
 
+			// Get children only for specific outputers
+			// in order to avoid slower status process
+			switch out.(type) {
+			case *dotOutput:
+				r := filepath.Join(p.AbsRoot, "vendor", string(proj.Ident().ProjectRoot))
+				ptr, err := gps.ListPackages(r, string(proj.Ident().ProjectRoot))
+
+				if err != nil {
+					return fmt.Errorf("analysis of %s package failed: %v", proj.Ident().ProjectRoot, err)
+				}
+
+				prm, _ := ptr.ToReachMap(true, false, false, nil)
+				bs.Children = prm.Flatten(false)
```

Comment:
There's some filtering that needs to be done here. Simply calling `Flatten` will get you the imports of ALL the packages in a ptree. We need only the subset of imports that come from the packages we're actually importing from a project.

The lock should contain the imported package list for each project, though, so this shouldn't be that bad to do.